community/values/src/main/java/org/neo4j/values/storable/LocalTimeValue.java

```diff
@@ -108,7 +109,16 @@ public static LocalTimeValue truncate(
             Supplier<ZoneId> defaultZone )
     {
         LocalTime localTime = input.getLocalTimePart();
-        LocalTime truncatedLT = localTime.truncatedTo( unit );
+        LocalTime truncatedLT;
+        try
+        {
+            truncatedLT = localTime.truncatedTo( unit );
+        }
+        catch ( UnsupportedTemporalTypeException e )
+        {
+            throw new UnsupportedTemporalUnitException( e.getMessage() );
```

Comment:
You should always pass in `e` as the cause (and thus make your Exception types accept an optional cause).